index.js

```diff
@@ -14,29 +14,42 @@
  */
 
 const debug = require('debug')('marinetrafficreporter')
-
 const Bacon = require('baconjs');
-
 const AisEncode = require("ggencoder").AisEncode
 const dgram = require('dgram')
+const _ = require('lodash')
+const util = require('util')
 
 module.exports = function(app) {
   var udpSocket
   var plugin = {}
   var unsubscribe = undefined
+  var timeout = undefined
 
   plugin.start = function(props) {
     debug("starting with " + props.ipaddress + ":" + props.port)
     var mmsi = app.config.settings.vessel.mmsi
 
+    if ( !mmsi )
+    {
+      console.log("marinetrafficreporter: mmsi missing in settings");
+      return
+    }
+    
     try {
       udpSocket = require('dgram').createSocket('udp4')
       unsubscribe = Bacon.combineWith(function(position, sog, cog, head) {
         return createPositionReportMessage(mmsi, position.latitude, position.longitude, mpsToKn(sog), cog, head)
       }, ['navigation.position', 'navigation.speedOverGround', 'navigation.courseOverGroundTrue', 'navigation.headingTrue'].map(app.streambundle.getSelfStream, app.streambundle)).changes().debounceImmediate((props.updaterate || 60)*1000).onValue(msg => {
 
-        sendPositionReportMsg(msg, props.ipaddress, props.port)
+        sendReportMsg(msg, props.ipaddress, props.port)
       })
+
+      timeout = setInterval(function() {
+        info = getStaticInfo()
```

Comment:
Sorry, I should have been clearer: I thought that the condition was ok, but that it should be done periodically, and not just once and give up forever.